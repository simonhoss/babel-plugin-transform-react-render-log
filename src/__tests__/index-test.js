const babel = require("babel-core");
const plugin = require("../");

var example = `
import React, {Component} from 'react'; 
export default class MyComp extends Component {
  render() {
    return <View></View>;
  } 
}`;

describe("when react preset", () => {
  it("should transform with console.log", () => {
    const { code } = babel.transform(example, {
      presets: ["react"],
      plugins: [plugin]
    });
    expect(code).toMatchSnapshot();
  });
});

describe("when react-native preset", () => {
  it("should transform with console.log", () => {
    const { code } = babel.transform(example, {
      presets: ["react-native"],
      plugins: [plugin]
    });
    expect(code).toMatchSnapshot();
  });
});
