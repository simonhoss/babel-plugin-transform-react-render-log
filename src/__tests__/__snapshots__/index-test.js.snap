// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when react preset should transform with console.log 1`] = `
"
import React, { Component } from 'react';
export default class MyComp extends Component {
  render() {
    console.group('MyComp render');
    console.log(this.props);
    console.log(this.state);
    console.groupEnd('render');

    return React.createElement(View, null);
  }
}"
`;

exports[`when react-native preset should transform with console.log 1`] = `"Object.defineProperty(exports,\\"__esModule\\",{value:true});var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if(\\"value\\"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _react=require('react');var _react2=_interopRequireDefault(_react);function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError(\\"Cannot call a class as a function\\");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError(\\"this hasn't been initialised - super() hasn't been called\\");}return call&&(typeof call===\\"object\\"||typeof call===\\"function\\")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!==\\"function\\"&&superClass!==null){throw new TypeError(\\"Super expression must either be null or a function, not \\"+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var MyComp=function(_Component){_inherits(MyComp,_Component);function MyComp(){_classCallCheck(this,MyComp);return _possibleConstructorReturn(this,(MyComp.__proto__||Object.getPrototypeOf(MyComp)).apply(this,arguments));}_createClass(MyComp,[{key:'render',value:function render(){console.group('MyComp render');console.log(this.props);console.log(this.state);console.groupEnd('render');return _react2.default.createElement(View,null);}}]);return MyComp;}(_react.Component);exports.default=MyComp;"`;
